src/ConnectivityWizard/Endpoints/ScanEndpoint.ts

```diff
@@ -0,0 +1,194 @@
+  private repeaterProcess: child_processes.ChildProcess;
+
+  constructor(tokenOps: TokensOperations) {
+    this.tokenOperations = tokenOps;
+  }
+
+  public async handle(ctx: Koa.Context): Promise<void> {
+    const req = <ScannedUrl>ctx.request.body;
+    const tokens: Tokens = this.tokenOperations.readTokens();
+    let scan_id: string = null;
+
+    if (!this.executeRepeater(tokens)) {
+      ctx.throw('Could not execute Repeater');
+
+      return;
+    }
+
+    try {
+      scan_id = await this.launchScan(req.url, tokens);
+    } catch (err) {
+      ctx.status = 400;
+
+      return;
+    }
+
+    ctx.body = <ScanId>{
+      scanId: scan_id
+    };
+  }
+
+  private executeRepeater(tokens: Tokens): boolean {
+    if (this.repeaterProcess != null && this.repeaterProcess.exitCode == null) {
+      logger.debug(
+        `Repeater process is still runnning (PID ${this.repeaterProcess.pid}). Skippiing spawning another one`
+      );
+
+      return true;
+    }
+
+    const node_exec = this.getNodeExec();
+
+    try {
+      const startArgs: string[] = [
+        ...node_exec.argv,
+        'repeater',
+        '--token',
+        `"${tokens.authToken}"`,
+        '--agent',
+        `"${tokens.repeaterId}`
+      ];
+      logger.debug(
+        `Launching Repeater process with cmd: ${
+          node_exec.cmd
+        } and arguments: ${JSON.stringify(startArgs)}`
+      );
+
+      this.repeaterProcess = child_processes.spawn(node_exec.cmd, startArgs, {
+        detached: true
+      });
+
+      this.repeaterProcess.stdout.on('data', (data) => {
+        const line = data.toString();
+        logger.debug(`Repeater (stdout): ${line}`);
+      });
+      this.repeaterProcess.stderr.on('data', (data) => {
+        const line = data.toString();
+        logger.error(`Repeater (stderr): ${line}`);
+      });
+
+      this.repeaterProcess.unref();
+
+      let fired = false;
+
+      this.repeaterProcess.on('close', (code, signal) => {
+        !fired &&
+          logger.log(
+            `Repeater process closed with exit code ${code} due to ${signal} signal`
+          );
+        fired = true;
+      });
+      this.repeaterProcess.on('error', (err: Error) => {
+        !fired &&
+          logger.log(`Failed to start repeater process due to ${err.message}`);
+        fired = true;
+      });
+      this.repeaterProcess.on('exit', (code) => {
+        !fired && logger.log(`Repeater process exited with exit code ${code}`);
+        fired = true;
+      });
+      logger.log(`Launched Repeater process (PID ${this.repeaterProcess.pid})`);
+
+      return true;
+    } catch (err) {
+      logger.error(`Failed to launch Repeater process. Error: ${err.message}`);
+
+      return false;
+    }
+  }
+
+  private async launchScan(url: string, tokens: Tokens): Promise<string> {
+    const nodeExec = this.getNodeExec();
+    const args: string[] = [
+      ...nodeExec.argv,
+      'scan:run',
+      '--token',
+      `"${tokens.authToken}"`,
```

Comment:
Double quote is redundant. Also, we should sanitize user input to prevent shell injections.

Reply:
quote removed. pending @bararchy response on token and repeater id valid syntax for validation.

Reply:
It might look like this (rfc822):
```
bchar = <any base64 character according to RFC 4648>
prefix-length = 7
hash-length = 32
prefix = prefix-length * bchar
type = "a" / "p" 
hash = hash-length * bchar
key = (prefix "." ("nex" type) "." hash)
```

e.g. `sdetod0.nexp.ayr9cm5szyaxxwaccb0egas0rdf6ltfd`